### PR TITLE
Hand evaluation

### PR DESCRIPTION
**Pull request by Ikshar**

### CODE REVIEW

File: src/logic/evaluateCombo.ts [hunk 1]
```diff
@@ -1,21 +1,189 @@
...
+
+  for (const die of dice) {
+    countMap[die] = (countMap[die] || 0) + 1;
+  }
+
+  return countMap;
+}
+
+export function evaluateCombo(dice: number[]): Scores {
+  const emptyScore = {
```
Comment (paleika) [inline src/logic/evaluateCombo.ts:40]: This object is stable so it's better to move it out of the function scope

File: src/logic/evaluateCombo.ts [hunk 2]
```diff
@@ -1,21 +1,189 @@
...
+  FullHouse: number;
+  SmallStraight: number;
+  LargeStraight: number;
+  Yahtzee: number;
+  Chance: number;
+}
+
+type Scores = UpperSection & LowerSection;
+
+enum FixedPayouts {
```
Comment (paleika) [inline src/logic/evaluateCombo.ts:22]: You're using this enum as a value only (never as a type), so it's better to make it a const object, no need for typescript here

File: src/components/App.tsx [hunk 3]
```diff
@@ -4,15 +4,16 @@ import { ControlButton } from "./ControlButton";
 import { DiceContainer } from "./DiceContainer";
 import { ScoreTable } from "./ScoreTable";
 import { TableContextProvider } from "../ctx/TableContext";
-import { RoundStage as RoundStage } from "../types";
+import { GameState, RoundStage as RoundStage } from "../types";
```
[the remark this answers is not in the document]
Comment (Ikshar) [reply src/components/App.tsx:7]: nope, no idea how it happened

File: src/components/ControlButton.tsx [hunk 4]
```diff
@@ -1,59 +1,83 @@
...
-    animateDice();
-    dispatch({ type: "setSelected" });
-    dispatch({ type: "setStage", payload: stageInfo[stage].nextStage });
-    dispatch({
-      type: "setValues",
-      payload: rerollUnheld(values, selected),
-    });
-    if (stage === RoundStage.Outcome) {
+    // todo: do not confirm on first roll
+    if (stage === RoundStage.Scoring || selectedRecord) {
```
[the remark this answers is not in the document]
Comment (Ikshar) [reply src/components/ControlButton.tsx:63]: The idea is that user can change their mind on adding a value to the table and unselect it, staying in the same stage

[the remark this answers is not in the document]
Comment (Ikshar) [reply src/components/ControlButton.tsx:63]: MB a better way to put it -- stage defines possible user actions: `roll, roll/score, roll/score, score`, accordingly

**Pull request merged**